docs/reference-use-cases.md

```diff
@@ -0,0 +1,215 @@
+
+- As an operator, given that I have deployed a cluster using Cluster API, I want to configure the cluster-autoscaler to drive scaling operations.
+
+- As an operator, given that I have a management cluster and a workload cluster, I want to retrieve, set, and change the number of worker Nodes or control plane Nodes in my workload cluster.
+
+- As an operator, given I have a management cluster and a workload cluster, I want to control the sizing, scaling, and optimizing of the workload cluster’s control plane in terms of Kubernetes primitives (e.g. HPA, VPA, resource limits, etc). I would like to import the best practices, sizing metrics and knowledge from e.g. the specialized SIG Scalability; the information should be expressed uniformly in Kubernetes terms.
+
+- As an operator, I expect the Cluster API to maintain the number and type of Nodes that I have currently requested as members of the cluster.
+
+### Configuration Updates
+
+- As an operator, given that I have a management cluster and a workload cluster, I want to update the IaaS credentials used to lifecycle manage my workload cluster because the correct credentials have changed.
+
+- As an operator, given I have a management cluster and a workload cluster, I want to change the “minor customizations” that I’m applying before kubelet starts.
+
+- As an operator, given that I have deployed a workload cluster via Cluster API, I want to change config in my workload cluster for which Cluster API is authoritative and have a Cluster API controller manage the deployment of that new configuration over the workload cluster.
+
+- As an operator, given that I have deployed a workload cluster via Cluster API and used Cluster API controller to manage the deployment of a new (broken) configuration, I want to revert to the previous working configuration.
+
+- As an operator, I want to declare the attributes (os, kernel, CRI)  of the nodes I want my workload to run on. The CAPI provider/controller should select an appropriate image that satisfies my constraints/attributes.
+    - If the provider does not support the attributes I have specified or find an appropriate image it should fail with an appropriate error.
+
+### Security
+
+- As an operator, given I have a management cluster and a workload cluster, I want to know when my cluster’s/machines’ certificates will expire, so that I can plan to rotate them.
+
+- As an operator, given I have a management cluster and a workload cluster, I want to automatically, periodically repave all the nodes of my cluster to reduce the risk of unauthorized software running on my machines.
+
+- As an operator, I want an external CA to sign certificates for the workload cluster control plane.
+
+- 🔭As an operator, given I have a management cluster and a workload cluster, I want to rotate all the certificates and credentials my machines are using.
+    - Some certificates might get rotated as part of machine upgrade, but are otherwise the above is out of scope.
```

Comment:
This comment makes it seem like the use case is invalid. Is it? I wonder if the either use case of the comment should be removed.

Reply:
I *believe* it was an attempt to clarify a case where rotating certificates might not be out of scope, compared to the general case where it would be out of scope. That said, it would definitely need more clarity and refinement when transferred to a CAEP